Framework/Algorithms/test/PolarizationCorrections/HeliumAnalyserEfficiencyTest.h

```diff
@@ -0,0 +1,280 @@
+    heliumAnalyserEfficiency->setProperty("InputWorkspace", groupWs->getName());
+    TS_ASSERT_THROWS(heliumAnalyserEfficiency->execute(), const std::runtime_error &);
+
+    MatrixWorkspace_sptr ws3 = generateWorkspace("ws3", x, y);
+    groupWs = groupWorkspaces("grp", std::vector<MatrixWorkspace_sptr>{ws1, ws2, ws3});
+    heliumAnalyserEfficiency->initialize();
+    heliumAnalyserEfficiency->setProperty("InputWorkspace", groupWs->getName());
+    TS_ASSERT_THROWS(heliumAnalyserEfficiency->execute(), const std::runtime_error &);
+
+    MatrixWorkspace_sptr ws4 = generateWorkspace("ws4", x, y);
+    groupWs = groupWorkspaces("grp", std::vector<MatrixWorkspace_sptr>{ws1, ws2, ws3, ws4});
+    heliumAnalyserEfficiency->initialize();
+    heliumAnalyserEfficiency->setProperty("InputWorkspace", groupWs->getName());
+    TS_ASSERT_THROWS_NOTHING(heliumAnalyserEfficiency->execute());
+  }
+
+  void testOutputs() {
+    auto wsGrp = createExampleGroupWorkspace("wsGrp");
+
+    auto heliumAnalyserEfficiency = AlgorithmManager::Instance().create("HeliumAnalyserEfficiency");
+    heliumAnalyserEfficiency->initialize();
+    heliumAnalyserEfficiency->setProperty("InputWorkspace", wsGrp->getName());
+    heliumAnalyserEfficiency->execute();
+
+    const std::string t = "T";
+    const std::string pHe = "p_He";
+    const std::string tPara = "T_para";
+    const std::string tAnti = "T_anti";
+
+    ASSERT_FALSE(AnalysisDataService::Instance().doesExist(t));
+    ASSERT_FALSE(AnalysisDataService::Instance().doesExist(pHe));
+    ASSERT_FALSE(AnalysisDataService::Instance().doesExist(tPara));
+    ASSERT_FALSE(AnalysisDataService::Instance().doesExist(tAnti));
+
+    heliumAnalyserEfficiency->initialize();
+    heliumAnalyserEfficiency->setProperty("InputWorkspace", wsGrp->getName());
+    heliumAnalyserEfficiency->setProperty("OutputTransmissionWorkspace", t);
+    heliumAnalyserEfficiency->setProperty("HeliumPolarisation", pHe);
+    heliumAnalyserEfficiency->setProperty("OutputTransmissionParaWorkspace", tPara);
+    heliumAnalyserEfficiency->setProperty("OutputTransmissionAntiWorkspace", tAnti);
+    heliumAnalyserEfficiency->execute();
+
+    ASSERT_TRUE(AnalysisDataService::Instance().doesExist(t));
+    AnalysisDataService::Instance().remove(t);
+    ASSERT_TRUE(AnalysisDataService::Instance().doesExist(pHe));
+    AnalysisDataService::Instance().remove(pHe);
+    ASSERT_TRUE(AnalysisDataService::Instance().doesExist(tPara));
+    AnalysisDataService::Instance().remove(tPara);
+    ASSERT_TRUE(AnalysisDataService::Instance().doesExist(tAnti));
+    AnalysisDataService::Instance().remove(tAnti);
+
+    auto members = wsGrp->getNames();
+    AnalysisDataService::Instance().remove(wsGrp->getName());
+    for (size_t i = 0; i < members.size(); ++i) {
+      AnalysisDataService::Instance().remove(members[i]);
+    }
+
+    AnalysisDataService::Instance().remove(heliumAnalyserEfficiency->getProperty("OutputWorkspace"));
```

Comment:
It's great to have `AnalysisDataService::Instance().clear()` in the test teardown method, so are all of these individual removes needed? Also, is there a way we could check some of the contents of the output workspaces to verify that they have been created successfully and pick up any regressions in future?

Reply:
I changed this test to just check for the workspaces existing in the ADS, instead of removing them as it goes, since they're going to get cleared out in the tear down anyway.